plugin/session_buffer.py

```diff
@@ -329,23 +330,28 @@ def purge_changes_async(self, view: sublime.View) -> None:
         finally:
             self._pending_changes = None
         self.session.notify_plugin_on_session_buffer_change(self)
-        sublime.set_timeout_async(lambda: self._on_after_change_async(view, version))
+        sublime.set_timeout_async(lambda: self._on_after_change_async(view, version, suppress_requests))
 
-    def _on_after_change_async(self, view: sublime.View, version: int) -> None:
+    def _on_after_change_async(self, view: sublime.View, version: int, suppress_requests: bool = False) -> None:
         if self._is_saving:
             self._has_changed_during_save = True
             return
-        self._do_color_boxes_async(view, version)
-        self.do_document_diagnostic_async(view, version)
-        if self.session.config.diagnostics_mode == "workspace" and \
-                not self.session.workspace_diagnostics_pending_response and \
-                self.session.has_capability('diagnosticProvider.workspaceDiagnostics'):
-            self._workspace_diagnostics_debouncer_async.debounce(
-                self.session.do_workspace_diagnostics_async, timeout_ms=WORKSPACE_DIAGNOSTICS_TIMEOUT)
-        self.do_semantic_tokens_async(view)
-        if userprefs().link_highlight_style in ("underline", "none"):
-            self._do_document_link_async(view, version)
-        self.do_inlay_hints_async(view)
+        if suppress_requests:
+            return
+        try:
+            self._do_color_boxes_async(view, version)
+            self.do_document_diagnostic_async(view, version)
+            if self.session.config.diagnostics_mode == "workspace" and \
+                    not self.session.workspace_diagnostics_pending_response and \
+                    self.session.has_capability('diagnosticProvider.workspaceDiagnostics'):
+                self._workspace_diagnostics_debouncer_async.debounce(
+                    self.session.do_workspace_diagnostics_async, timeout_ms=WORKSPACE_DIAGNOSTICS_TIMEOUT)
+            self.do_semantic_tokens_async(view)
+            if userprefs().link_highlight_style in ("underline", "none"):
+                self._do_document_link_async(view, version)
+            self.do_inlay_hints_async(view)
+        except MissingUriError:
```

Comment:
Before I forget again, 
was this an intentional addition?
```
except MissingUriError:
  pass
```

Reply:
Yes, I've added the `try` block for the requests because this function is put on the async thread via `sublime.set_timeout_async` at the end of `purge_changes_async`. So it could theoretically happen that the view has closed in the meantime, and then it would raise the MissingUriError. In that case nothing else needs to happen; we could also use `return` here but I used `pass` because this is at the end of the function anyway.